components/Posts.php

```diff
@@ -206,11 +215,26 @@ protected function listPosts()
         /*
          * Add a "url" helper attribute for linking to each post and category
          */
-        $posts->each(function($post) {
-            $post->setUrl($this->postPage, $this->controller);
-
-            $post->categories->each(function($category) {
-                $category->setUrl($this->categoryPage, $this->controller);
+        $blogPostComponent = $this->getComponent('blogPost', $this->postPage);
+        $blogPostsComponent = $this->getComponent('blogPosts', $this->categoryPage);
+
+        $posts->each(function($post) use ($blogPostComponent, $blogPostsComponent) {
```

Comment:
@GinoPane space between function and arguments as per PSR-2 guidelines.
```suggestion
        $posts->each(function ($post) use ($blogPostComponent, $blogPostsComponent) {
```